### PR TITLE
Changes in Entry Point

### PR DESCRIPTION
**Pull request by ahilles107**
Now you can set properly redirect url

### CODE REVIEW

File: Security/EntryPoint/FacebookAuthenticationEntryPoint.php [hunk 1]
```diff
@@ -46,14 +46,15 @@ class FacebookAuthenticationEntryPoint implements AuthenticationEntryPointInterf
...
                 'scope' => implode(',', $this->permissions),
-                'redirect_uri' => $request->getUriForPath($this->options->get('check_path', '')),
+                'redirect_uri' => str_replace($this->options->get('server_url'), $this->options->get('app_url'), $request->getUriForPath($this->options->get('check_path', ''))),
             ))
+            
         );
-
-        return $response;
+		echo '<script>top.location.href="'.$loginUrl.'"</script>';
+        die;
```
Comment (lsmith77) [inline Security/EntryPoint/FacebookAuthenticationEntryPoint.php]: you should in this case not die. instead you should check if the user is doing a canvas app by looking at the options. if the user does have a canvas app you should use a normal `Reponse` instance and set what you are echoing as the content. Though I think it should be valid html, aka it should have an `<html>` tag etc.

Comment (ahilles107) [inline Security/EntryPoint/FacebookAuthenticationEntryPoint.php]: ok

File: Security/EntryPoint/FacebookAuthenticationEntryPoint.php [hunk 2]
```diff
@@ -46,14 +46,18 @@ class FacebookAuthenticationEntryPoint implements AuthenticationEntryPointInterf
...
                 'display' => $this->options->get('display', 'page'),
                 'scope' => implode(',', $this->permissions),
-                'redirect_uri' => $request->getUriForPath($this->options->get('check_path', '')),
+                'redirect_uri' => str_replace($this->options->get('server_url'), $this->options->get('app_url'), $request->getUriForPath($this->options->get('check_path', ''))),
             ))
+            
         );
-
+        
+        if($this->options->get('server_url') & $this->options->get('app_url'))
```
Comment (lsmith77) [inline Security/EntryPoint/FacebookAuthenticationEntryPoint.php]: please follow the Symfony2 CS. aka a space after `if` and 4 spaces for indenting

File: Security/EntryPoint/FacebookAuthenticationEntryPoint.php [hunk 3]
```diff
@@ -46,14 +46,18 @@ class FacebookAuthenticationEntryPoint implements AuthenticationEntryPointInterf
      */
     public function start(Request $request, AuthenticationException $authException = null)
     {
-        $response = new RedirectResponse($this->facebook->getLoginUrl(
+        $response = new RedirectResponse($loginUrl = $this->facebook->getLoginUrl(
```
Comment (lsmith77) [inline Security/EntryPoint/FacebookAuthenticationEntryPoint.php]: there is no reason to create the `RedirectResponse` in the case when its a canvas app, more over if its not .. then the URL shouldnt be changed.

File: Security/EntryPoint/FacebookAuthenticationEntryPoint.php [hunk 4]
```diff
@@ -46,14 +46,23 @@ class FacebookAuthenticationEntryPoint implements AuthenticationEntryPointInterf
      */
     public function start(Request $request, AuthenticationException $authException = null)
     {
-        $response = new RedirectResponse($this->facebook->getLoginUrl(
+        if ($this->options->get('server_url') & $this->options->get('app_url')){
```
Comment (lsmith77) [inline Security/EntryPoint/FacebookAuthenticationEntryPoint.php]: i assume this is supposed to be `&&`
also there should be a space after the closing bracket and before `{`

so the line should be:
`if ($this->options->get('server_url') && $this->options->get('app_url')) {`

same applies below

File: Security/EntryPoint/FacebookAuthenticationEntryPoint.php [hunk 5]
```diff
@@ -46,14 +46,23 @@ class FacebookAuthenticationEntryPoint implements AuthenticationEntryPointInterf
...
+            $redirect_uri = str_replace($this->options->get('server_url'), $this->options->get('app_url'), $request->getUriForPath($this->options->get('check_path', '')));
+        } else {
+            $redirect_uri = $request->getUriForPath($this->options->get('check_path', ''));
+        }
+
```
Comment (lsmith77) [inline Security/EntryPoint/FacebookAuthenticationEntryPoint.php:54]: i would suggest restructuring the code for better readability:

```
$redirect_uri = $request->getUriForPath($this->options->get('check_path', ''));
if ($this->options->get('server_url') && $this->options->get('app_url')) {
    $redirect_uri = str_replace($this->options->get('server_url'), $this->options->get('app_url'), $redirect_uri);
}
```

**Pull request merged**